modules/@apostrophecms/settings/index.js

```diff
@@ -0,0 +1,627 @@
+//     label: 'Account',
+//     // The subforms that belong to the group. The order is respected.
+//     subforms: [ 'name', 'password' ]
+//   }
+// }
+// ```
+//
+// ## API
+//
+// Add a protected field to the system protected fields list. This will ensure
+// that any subform containing that field will be ALWAYS protected by
+// the user's current password. It is recommended to use this method in the
+// `apostrophe:modulesRegistered` event handler.
+// `self.apos.settings.addProtectedField(fieldName, protectionType)`
+//
+// Add a forbidden field to the forbidden fields list. This will ensure that
+// the field will not be allowed in any subform. It is recommended to use this
+// method in the `apostrophe:modulesRegistered` event handler.
+// `self.apos.settings.addForbiddenField(fieldName)`
+//
+// Add a field to the reload after save fields list. This will ensure that
+// the subform containing the field will be reloaded after save.
+// It is recommended to use this method in the `apostrophe:modulesRegistered`
+// event handler.
+// `self.apos.settings.addReloadAfterSaveField(fieldName)`
+//
+// ## UI
+//
+// An example of a custom `previewComponent` with the core components explained
+// can be found in the relevant PR:
+// https://github.com/apostrophecms/apostrophe/pull/4236
+const { klona } = require('klona');
+
+module.exports = {
+  options: {
+    alias: 'settings',
+    subforms: {},
+    groups: {}
+  },
+
+  init(self) {
+    // List of all allowed protection types and their aliases (`subform.protection: type`).
+    // The key is the type or alias, the value is the actual type (always a string).
+    // All subforms `protection` prop will be converted to the actual type.
+    // Invalid protection type will panic.
+    self.protectionTypes = {
+      // Protection type to be used if protected is simply set to `true`.
+      true: 'password',
+      password: 'password'
+      // TODO phase 3
+      // email: 'email'
+    };
+    // Collection of fieldName: protectionType objects for system forced protected fields.
+    // The order is important, the first match is used (first have higher priority).
+    // If there are multiple fields in the subform, having a system protected field,
+    // the first match from this list wins. If there is specifically `password` field
+    // in the subform, the schema will be completely replaced with the auto-generated
+    // password schema.
+    // Do not modify this object directly, use
+    // `self.apos.settings.addProtectedField(fieldName, protectionType)` instead.
+    self.systemProtectedFields = {
+      password: self.protectionTypes.password
+      // TODO phase 3
+      // username: self.protectionTypes.password,
+      // email: self.protectionTypes.email
+    };
+    // Completely forbidden fields, they are not allowed in the subforms.
+    // Do not modify this array directly, use
+    // `self.apos.settings.addForbiddenField(fieldName)` instead.
+    self.forbiddenFields = [
```

Comment:
system*

Reply:
I'm not following. `systemForbiddenFields`?